### ISSUE TITLE
querying changes stream from npm regsitry reports error

### ISSUE THREAD

**Issue by sify21** [mibes/couch-rs#11]
https://replicate.npmjs.com exposes couchdb's `_changes` api. when using this library to query it, it reports error
```
CouchError { id: None, status: 501, message: "data did not match any variant of untagged enum Event" }
```
The code is
```
use anyhow::Result;
use futures_util::stream::StreamExt;

#[tokio::main]
async fn main() -> Result<()> {
    let client = couch_rs::Client::new_no_auth("https://replicate.npmjs.com")?;
    let db = client.db("").await?;
    let mut changes = db.changes(None);
    changes.set_infinite(true);
    let mut events = vec![];
    while let Some(change) = changes.next().await {
        match change {
            Ok(c) => {
                events.push(c);
                if events.len() == 10 {
                    break;
                }
            }
            Err(e) => println!("error: {:?}", e),
        }
    }
    println!("{:?}", &events);
    Ok(())
}
```

**Comment by mibes** [comment]
It looks like the `_changes` stream from https://replicate.npmjs.com returns the seq as a number:

```json
   {"results":[
     {"seq":10334,"id":"scaphold-cli","changes":[{"rev":"1-f0a432ce49511bd26e28c51691f97642"}]},
     {"seq":10336,"id":"@pod-point/eslint-config-podpoint","changes":[{"rev":"1-aa419e700c36aa3fbb8b0e34e5168e48"}]}
    ...
```

On our reference system seq is a string:
```json
    {"results":[
       {"seq":"4-g1AAAAB8eJzLYWBgYMpgTmEQSs4vTc5ISXIwNNAzBEFTkxyQXCJDUv3___-zMpgTWXKBAuwWxolppqkGWHXgMyiPBUgyNACp_yDzEhmyAK3dIP4","id":"station_location_wl_neu_orga_o_tk_2","changes":[{"rev":"1-c743165640469df69d867b016d3fba27"}]},
       {"seq":"9-g1AAAACfeJzLYWBgYMpgTmEQSs4vTc5ISXIwNNAzBEFTkxyQXCJDUv3___-zMpgTWXKBAuwWxolppqkGWHXgMyiPBUgyNACp_1DzWMHmpVlYGpglpWDVmAUAvF8pWg","id":"station_location_wl_neu_orga_o_tk_1","changes":[{"rev":"1-7e2f057950e1709e57c0af71ed49426a"}]},
...
```

According to the spec (https://docs.couchdb.org/en/main/api/database/changes.html?highlight=_changes), `seq` is a json, so both should be possible: 
```
seq (json) – Update sequence.
```

Currently the code presumes it is a `string`. 

Let's see if changing this for `serde_json::Value` is enough to fix it.

**Comment by mibes** [comment]
That looks to be working! Let me run a few more tests tomorrow to see if there are unwanted side-effects.

**Comment by mibes** [comment]
Included the fix in the 0.8.32 release. Will close this issue for now.
